pkg/git/git.go

```diff
@@ -587,7 +610,25 @@ func (r *Repo) Push(remoteBranch string) error {
 	}
 	args = append(args, DefaultRemote, remoteBranch)
 
-	return command.NewWithWorkDir(r.Dir(), gitExecutable, args...).RunSuccess()
+	if err := command.NewWithWorkDir(r.Dir(), gitExecutable, args...).RunSuccess(); err != nil {
+		return NewNetworkError(err)
+	}
+	return nil
+}
+
+// PushWithRetries tries to push a branch, retrying when the error allows it
```

Comment:
Really cool to see this new functionality @puerco! One thing I was wondering was how often we think there may be additional `___WithRetries` methods? If we think there will be more than this one (seems likely), what would you think about making the retry configuration part of `Repo` itself, similar to what we do with `dryRun`, etc.?

Reply:
Good point @hasheddan , I'm going to modify it that way and we can start adding retries to the git functions as we need them :+1:

Reply:
OK, I've added the retry capability to the `git.Repo` object. It now has a new method to define the number of retries:

`repo.SetMaxRetries()`